pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
Right, we can't force every client to ignore the NotFound error. I'll think harder...